plugins/kubernetes-backend/src/service/KubernetesProxy.ts

```diff
@@ -0,0 +1,286 @@
+    details: ClusterDetails,
+    req: Request,
+  ): Promise<KubernetesProxyResponse> {
+    const serverIP = this.getClusterURI(details);
+    if (!serverIP) {
+      this.logger.error(
+        `could not find server IP for cluster with name ${details.name}`,
+      );
+      return {
+        code: ERROR_INTERNAL_SERVER,
+        data: null,
+      };
+    }
+
+    const query = decodeURIComponent(req.params.encodedQuery) || '';
+    const uri = `${serverIP}/${query}`;
+
+    const contentType = req.header(HEADER_CONTENT_TYPE) || APPLICATION_JSON;
+
+    const res = await this.sendClusterRequest(
+      details,
+      uri,
+      req.method,
+      contentType,
+      req.body,
+    );
+
+    return res;
+  }
+
+  protected async sendClusterRequest(
+    details: ClusterDetails,
+    uri: string,
+    method: string,
+    contentType: string,
+    body?: any,
+  ): Promise<KubernetesProxyResponse> {
+    const bearerToken = details.serviceAccountToken;
+    if (!bearerToken) {
+      this.logger.error('could not find bearer token!');
+      return {
+        code: ERROR_BAD_REQUEST,
+        data: {
+          error: 'Invalid service account token',
+        },
+      };
+    }
+
+    const reqData: any = {
+      method,
+      headers: {
+        'Content-Type': contentType,
+        Authorization: `Bearer ${bearerToken}`,
+      },
+    };
+
+    if (!details.skipTLSVerify) {
+      if (details.caData) {
+        const ca = bufferFromFileOrString('', details.caData)?.toString() || '';
+        reqData.agent = new https.Agent({ ca });
+      } else {
+        this.logger.info('could not find CA certificate!');
+        return {
+          code: ERROR_INTERNAL_SERVER,
+          data: {
+            error: 'Invalid CA certificate configured within Backstage',
+          },
+        };
+      }
+    }
+
+    if (body && Object.keys(body).length > 0) {
+      reqData.body = JSON.stringify(body);
+    }
+
+    try {
+      const req = await fetch(uri, reqData);
+
+      let res;
+      if (contentType === APPLICATION_JSON) {
+        res = await req.json();
+      } else {
+        res = await req.text();
+      }
+
+      const proxyResponse: KubernetesProxyResponse = {
+        code: req.status,
+        data: res,
+        cluster: details.name,
+      };
+
+      return proxyResponse;
+    } catch (e: any) {
+      this.logger.error(`error with request: ${JSON.stringify(e)}`);
+      return {
+        code: ERROR_INTERNAL_SERVER,
+        data: e,
+        cluster: details.name,
+      };
+    }
+  }
+
+  protected getKubeConfig(clusterDetails: ClusterDetails): KubeConfig {
+    const cluster = {
+      name: clusterDetails.name,
+      server: clusterDetails.url,
+      skipTLSVerify: clusterDetails.skipTLSVerify,
+      caData: clusterDetails.caData,
+    };
+
+    const user = {
+      name: 'backstage',
+      token: clusterDetails.serviceAccountToken,
+    };
+
+    const context = {
+      name: `${clusterDetails.name}`,
```

Comment:
no need to interpolate this with backticks, right?

Reply:
Done